src/main/java/org/osc/manager/rest/server/api/SecurityGroupApis.java

```diff
@@ -0,0 +1,129 @@
+
+    private TransactionControl txControl;
+
+    private ValidationUtil validationUtil;
+
+    public void init(EntityManager em, TransactionControl txControl) throws Exception {
+        this.em = em;
+        this.txControl = txControl;
+        this.validationUtil = new ValidationUtil(txControl, em);
+    }
+
+    @POST
+    public String createSecurityGroup(@PathParam("deviceId") Long deviceId, SecurityGroupEntity entity)
+            throws Exception {
+        LOG.info(String.format("Creating security group  with name %s", entity.getName()));
+        DeviceEntity device = this.validationUtil.getDeviceOrThrow(Long.toString(deviceId));
+        this.validationUtil.validateParentIdMatches(device, Long.parseLong(entity.getDevice().getId()));
+        // TODO : SUDHIR - Add SecurityGroupMember
+        VSElement vs = new VSElement(deviceId, null);
+        this.sgApi = new IsmSecurityGroupApi(vs, this.txControl, this.em);
+        return this.sgApi.createSecurityGroup(entity.getName(), null, null);
+    }
+
+    @Path("/{sgId}")
+    @PUT
+    public SecurityGroupEntity updateSecurityGroup(@PathParam("deviceId") Long deviceId, @PathParam("sgId") Long sgId,
+            SecurityGroupEntity entity)
+                    throws Exception {
+        LOG.info(String.format("Updating the security group for id %s ", Long.toString(sgId)));
+        DeviceEntity device = this.validationUtil.getDeviceOrThrow(Long.toString(deviceId));
+        this.validationUtil.validateParentIdMatches(device, Long.parseLong(entity.getDevice().getId()));
+        // TODO : SUDHIR - Add SecurityGroupMember
+        VSElement vs = new VSElement(deviceId, null);
+        this.sgApi = new IsmSecurityGroupApi(vs, this.txControl, this.em);
+        this.sgApi.updateSecurityGroup(Long.toString(sgId), entity.getName(), null);
+        return entity;
+    }
+
+    @Path("/{sgId}")
+    @DELETE
+    public void deleteSecurityGroup(@PathParam("sgId") Long sgId, @PathParam("deviceId") Long deviceId)
+            throws Exception {
+        LOG.info(String.format("Deleting the security group for id %s ", Long.toString(sgId)));
+        this.validationUtil.getDeviceOrThrow(Long.toString(deviceId));
+        VSElement vs = new VSElement(deviceId, null);
+        this.sgApi = new IsmSecurityGroupApi(vs, this.txControl, this.em);
+        this.sgApi.deleteSecurityGroup(Long.toString(sgId));
+    }
+
+    @GET
+    public List<String> getSecurityGroupIds(@PathParam("deviceId") Long deviceId) throws Exception {
+        LOG.info("Listing security group ids'");
+        this.validationUtil.getDeviceOrThrow(Long.toString(deviceId));
+        VSElement vs = new VSElement(deviceId, null);
+        this.sgApi = new IsmSecurityGroupApi(vs, this.txControl, this.em);
+        List<? extends ManagerSecurityGroupElement> securityGroups = this.sgApi.getSecurityGroupList();
+        List<String> sgList = new ArrayList<String>();
+        if (!securityGroups.isEmpty()) {
```

Comment:
This check is redundant. Please see above block to optimize the 4 lines of code into one.